SnackOverflow/src/main/java/ro/utcn/sd/vasi/SnackOverflow/data_assembler/ScoreCalcuator.java

```diff
@@ -0,0 +1,42 @@
+package ro.utcn.sd.vasi.SnackOverflow.data_assembler;
+
+
+import ro.utcn.sd.vasi.SnackOverflow.model.VoteAnswer;
+import ro.utcn.sd.vasi.SnackOverflow.model.VoteQuestion;
+
+
+import java.util.List;
+
+public class ScoreCalcuator {
+
+    public static int computeUserScore(List<VoteQuestion> receivedQuestion, List<VoteAnswer> receivedAnswer, List<VoteQuestion> givenQuestion, List<VoteAnswer> givenAnsser) {
+        int nrReceivedUpvotedQuestions = 0, nrReceivedDownvotedQuestions = 0;
+        int nrReceivedUpvotedAnswers = 0, nrReceivedDownvotedAnswers = 0;
+        int nrGivedDownvotedPosts = 0;
+
+        nrReceivedUpvotedQuestions = (int)receivedQuestion.stream().filter(x->x.isValue()).count();
```

Comment:
You can replace the x->x.isValue() lambda with a method reference: `VoteQuestion::isValue`.